File: easy.server.demo/easy.server.boot.demo/src/main/java/io/github/jinlongliao/easy/server/boot/demo/core/parser/DemoMessageParserCallBack.java
package io.github.jinlongliao.easy.server.boot.demo.core.parser;

import io.github.jinlongliao.easy.server.core.parser.IMessageParserCallBack;
import io.github.jinlongliao.easy.server.core.parser.IRequestStreamFactory;
import io.github.jinlongliao.easy.server.core.parser.MeType;
import io.github.jinlongliao.easy.server.mapper.spring.IBeanMapper;

import jakarta.servlet.http.HttpServletRequest;

/**
 * @author: liaojinlong
 * @date: 2022-09-16 10:34
 */
public class DemoMessageParserCallBack implements IMessageParserCallBack {
    private final IBeanMapper beanMapper;

    public DemoMessageParserCallBack(IBeanMapper beanMapper) {
        this.beanMapper = beanMapper;
    }

    @Override
    public Object parserParamBody(IRequestStreamFactory request, MeType meType, Object... args) {
        throw new UnsupportedOperationException();
    }

    @Override
    public Object parserParamBody(HttpServletRequest request, MeType meType, Object... args) {
        return this.beanMapper.servletBeanMapper(meType.getType(), request);
    }

    @Override
    public Object parserCommonParam(IRequestStreamFactory request, MeType meType, Object... args) {
        throw new UnsupportedOperationException();
    }

    @Override
    public Object parserCommonParam(HttpServletRequest request, MeType meType, Object... args) {
        throw new UnsupportedOperationException();
    }


}

File: easy.server.demo/easy.server.kotlin.demo/src/main/kotlin/io/github/jinlongliao/easy/server/kotlin/demo/core/parser/DemoMessageParserCallBack.kt
package io.github.jinlongliao.easy.server.kotlin.demo.core.parser

import io.github.jinlongliao.easy.server.core.parser.IMessageParserCallBack
import io.github.jinlongliao.easy.server.core.parser.IRequestStreamFactory
import io.github.jinlongliao.easy.server.core.parser.MeType
import io.github.jinlongliao.easy.server.mapper.spring.IBeanMapper
import jakarta.servlet.http.HttpServletRequest

/**
 * @author: liaojinlong
 * @date: 2022-09-16 10:34
 */
class DemoMessageParserCallBack(private val beanMapper: IBeanMapper) : IMessageParserCallBack {
    override fun parserParamBody(request: IRequestStreamFactory, meType: MeType, vararg args: Any): Any {
        throw UnsupportedOperationException()
    }

    override fun parserParamBody(request: HttpServletRequest, meType: MeType, vararg args: Any): Any {
        return beanMapper.servletBeanMapper(meType.type, request)
    }

    override fun parserCommonParam(request: IRequestStreamFactory, meType: MeType, vararg args: Any): Any {
        throw UnsupportedOperationException()
    }

    override fun parserCommonParam(request: HttpServletRequest, meType: MeType, vararg args: Any): Any {
        throw UnsupportedOperationException()
    }
}
